libraries/helpers.rb

```diff
@@ -46,15 +46,12 @@ def create_chef_directories
         d_owner = root_owner
         %w(run_path cache_path backup_path log_dir conf_dir).each do |dir|
           # Do not redefine the resource if it exist
-          begin
-            resources(directory: node['chef_client'][dir])
-          rescue Chef::Exceptions::ResourceNotFound
-            directory node['chef_client'][dir] do
-              recursive true
-              mode '0755' if dir == 'log_dir'
-              owner d_owner
-              group node['root_group']
-            end
+          next if find_resource(:directory, node['chef_client'][dir])
+          directory node['chef_client'][dir] do
```

Comment:
find_resource with a block actually does exactly what you want:

```
find_resource(:directory, node['chef_client'][dir]) do
            recursive true
            mode '0755' if dir == 'log_dir'
            owner d_owner
            group node['root_group']
end
```

no need to guard it.

https://github.com/chef/chef/blob/master/lib/chef/dsl/declare_resource.rb#L187-L190

Reply:
Updated. Thanks